src/dune_pkg/opam_solver.ml

```diff
@@ -450,12 +474,15 @@ let opam_package_to_lock_file_pkg ~repo ~local_packages opam_package =
     let build_step =
       opam_commands_to_actions opam_package (OpamFile.OPAM.build opam_file)
     in
-    List.concat [ subst_step; patch_step; build_step ] |> make_action
+    List.concat [ subst_step; patch_step; build_step ]
+    |> make_action
+    |> Option.map ~f:build_env
```

Comment:
You should add a test for an opam file that has no install/build and yet has `build-env`.

Reply:
Done.